File: Backend/uniguide/src/main/java/com/uniguide/controller/AppFeedbackController.java
package com.uniguide.controller;

import java.util.List;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.uniguide.beans.AppFeedback;
import com.uniguide.beans.Event;
import com.uniguide.beans.University;
import com.uniguide.beans.UserLogin;
import com.uniguide.service.AppFeedbackService;

@RequestMapping("/appfeedback")
@RestController
@CrossOrigin
public class AppFeedbackController {
	@Autowired
	private AppFeedbackService appfeedback;
	
	
	@PostMapping("/")
	public ResponseEntity<?> giveFeedback( @RequestBody AppFeedback fb) {
		System.out.println("inside the controller");
		System.out.println("feedback"+fb);
		appfeedback.giveFeedback(fb);
		  return ResponseEntity.ok("feedback added successfully");
	}
	
	@GetMapping("/")
	public List<AppFeedback> getFeedbackData() {
		System.out.println(appfeedback.getfeedback());
		return appfeedback.getfeedback();
		  
	}
	
	@PutMapping("/")
	public ResponseEntity<String> updateFeedback(@RequestBody AppFeedback fb){
		boolean status=appfeedback.update(fb);
		if(status)
			return ResponseEntity.ok("Feedback updated successfully");
		
		return ResponseEntity.ok("feedback not found");
	}
	
	@DeleteMapping("/{id}")
	public ResponseEntity<String> deleteFeedback(@PathVariable int id){
		boolean status=appfeedback.delete(id);
		if(status)
			return ResponseEntity.ok("feedback deleted succesfully");
		return ResponseEntity.ok("feedback deleted unsuccesfully");
	}
	
	@GetMapping("/recent")
	public List<AppFeedback> getRecentFeedback()
	{
		List<AppFeedback> feedlist= appfeedback.getRecentFeedbacks();
		return feedlist;
	}
	
}

File: .net/Controller/AppFeedBackController.cs
using Microsoft.AspNetCore.Mvc;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using YourNamespace.Models; // Replace with your actual model namespace
using YourNamespace.Services; // Replace with your actual service namespace

namespace YourNamespace.Controllers
{
    [Route("api/appfeedback")]
    [ApiController]
    public class AppFeedbackController : ControllerBase
    {
        private readonly IAppFeedbackService _appFeedbackService;

        public AppFeedbackController(IAppFeedbackService appFeedbackService)
        {
            _appFeedbackService = appFeedbackService;
        }

        [HttpPost]
        public IActionResult GiveFeedback([FromBody] AppFeedback fb)
        {
            if (fb == null)
            {
                return BadRequest("Invalid feedback data");
            }

            // Assuming that _appFeedbackService.giveFeedback() returns a boolean indicating success
            if (_appFeedbackService.GiveFeedback(fb))
            {
                return Ok("Feedback added successfully");
            }
            else
            {
                return BadRequest("Failed to add feedback");
            }
        }

        [HttpGet]
        public ActionResult<IEnumerable<AppFeedback>> GetFeedbackData()
        {
            var feedbackData = _appFeedbackService.GetFeedback();
            return Ok(feedbackData);
        }

        [HttpPut]
        public IActionResult UpdateFeedback([FromBody] AppFeedback fb)
        {
            if (fb == null)
            {
                return BadRequest("Invalid feedback data");
            }

            // Assuming that _appFeedbackService.update() returns a boolean indicating success
            if (_appFeedbackService.Update(fb))
            {
                return Ok("Feedback updated successfully");
            }
            else
            {
                return BadRequest("Feedback not found");
            }
        }

        [HttpDelete("{id}")]
        public IActionResult DeleteFeedback(int id)
        {
            // Assuming that _appFeedbackService.delete() returns a boolean indicating success
            if (_appFeedbackService.Delete(id))
            {
                return Ok("Feedback deleted successfully");
            }
            else
            {
                return BadRequest("Feedback not found");
            }
        }

        [HttpGet("recent")]
        public ActionResult<IEnumerable<AppFeedback>> GetRecentFeedback()
        {
            var recentFeedback = _appFeedbackService.GetRecentFeedbacks();
            return Ok(recentFeedback);
        }
    }
}
